odxtools/compumethods/tabintpcompumethod.py

```diff
@@ -2,28 +2,144 @@
+                    </COMPU-SCALE>
+                    <COMPU-SCALE>
+                        <LOWER-LIMIT INTERVAL-TYPE = "CLOSED">30</LOWER-LIMIT>
+                        <COMPU-CONST>
+                            <V>2</V>
+                        </COMPU-CONST>
+                    </COMPU-SCALE>
+                </COMPU-SCALES>
+            </COMPU-INTERNAL-TO-PHYS>
+        </COMPU-METHOD>
+
+    """
+
     def __init__(self,
-                 internal_type,
-                 physical_type):
+                 internal_type: Union[DataType, str],
+                 physical_type: Union[DataType, str],
+                 internal_points: List[Union[float, int]],
+                 physical_points: List[Union[float, int]]):
         super().__init__(internal_type, physical_type, "TAB-INTP")
-        logger.debug("Created table interpolation compu method!")
-        logger.warning(
-            "TODO: Implement table interpolation compu method properly!")
 
-    def convert_physical_to_internal(self, physical_value):
-        return self.internal_type.make_from(physical_value)
+        self.internal_points = internal_points
+        self.physical_points = physical_points
+
+        self._physical_lower_limit = Limit(
+            min(physical_points), IntervalType.CLOSED)
+        self._physical_upper_limit = Limit(
+            max(physical_points), IntervalType.CLOSED)
+
+        logger.debug("Created compu method of type tab interpolated !")
+        self._assert_validity()
+
+    @property
+    def physical_lower_limit(self):
+        return self._physical_lower_limit
+
+    @property
+    def physical_upper_limit(self):
+        return self._physical_upper_limit
+
+    def _assert_validity(self) -> None:
+        assert len(self.internal_points) == len(self.physical_points)
+
+        assert self.internal_type in [DataType.A_INT32, DataType.A_UINT32,
+                                      DataType.A_FLOAT32, DataType.A_FLOAT64], \
+            ("Internal data type of tab-intp compumethod must be one of"
+             " [DataType.A_INT32, DataType.A_UINT32, DataType.A_FLOAT32, DataType.A_FLOAT64]")
+        assert self.physical_type in [DataType.A_INT32, DataType.A_UINT32,
+                                      DataType.A_FLOAT32, DataType.A_FLOAT64], \
+            ("Physical data type of tab-intp compumethod must be one of"
+             " [DataType.A_INT32, DataType.A_UINT32, DataType.A_FLOAT32, DataType.A_FLOAT64]")
+
+    def _piecewise_linear_interpolate(self,
+                                      x: Union[int, float],
+                                      points: List[Tuple[Union[int, float], Union[int, float]]]) \
+            -> Union[float, int, None]:
+        for ((x0, y0), (x1, y1)) in zip(points[:-1], points[1:]):
+            if x0 <= x and x <= x1:
+                return y0 + (x - x0) * (y1 - y0) / (x1 - x0)
+
+        return None
+
+    def convert_physical_to_internal(self, physical_value: Union[int, float]):
```

Comment:
no type annotation for the return value. intentional? (same applies to a few other new functions.)

Reply:
Just me being sloppy. I added the type annotations.